src/main/java/org/openstreetmap/atlas/geography/boundary/CountryBoundaryMap.java

```diff
@@ -1212,7 +1223,10 @@ public int size()
         }
 
         // Performance: short circuit, if all intersected polygons in same country, skip cutting.
-        if (isSameCountry(candidates))
+        // (except when geometry has to be sliced at all costs)
+        if (isSameCountry(candidates)
+                && (sources == null || sources.length == 0 || this.forceSlicingPredicate == null
```

Comment:
It might be worth extracting the sources and slicing logic to a private method to avoid duplication. The multiple sources is a bit confusing as well, as @mertemin pointed out.

Reply:
Will do